infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -105,9 +105,9 @@ public LogReplicationConfig(LogReplicationConfigManager configManager,
                                 int maxNumMsgPerBatch, int maxMsgSize, int cacheSize, int maxSnapshotEntriesApplied) {
         this.configManager = configManager;
         this.maxNumMsgPerBatch = maxNumMsgPerBatch;
-        this.maxMsgSize = maxMsgSize;
         this.maxCacheSize = cacheSize;
-        this.maxDataSizePerMsg = maxMsgSize * DATA_FRACTION_PER_MSG / 100;
+        this.maxMsgSize = maxMsgSize;
+        this.maxDataSizePerMsg = (int) ((maxMsgSize * DATA_FRACTION_PER_MSG) / 100L);
```

Comment:
For this and the above static var as well, they're used in quite a bit of places. I agree that these along with the rest of our config values should be moved to longs for safety, and can for sure take that up with future cleanup. For this this change though to keep it to a minimum was thinking int should be fine since for the overflow to occur the param would have to be set to >2GB which would be exceeding our JVM xmx anyways

Reply:
int should suffice the usage intended for maxMsgSize.

Reply:
can you verify that (maxMsgSize * 90) does not result in an overflow even though maxMsgSize is an int?

Reply:
Yes, with `DATA_FRACTION_PER_MSG` being changed to a long the math will look like this since java uses the highest order for operations:

```
    public static void main(String[] args) {
        System.out.println(((64 << 20) * 90)); // 1744830464
        System.out.println(((64 << 20) * 90L)); // 6039797760
    }
```